nifi-extension-bundles/nifi-atlassian-bundle/nifi-atlassian-extensions/src/main/java/org/apache/nifi/atlassian/bitbucket/BitbucketRepositoryClient.java

```diff
@@ -299,13 +419,43 @@ public String createContent(final GitCreateContentRequest request) throws FlowRe
                     String.format("Error while committing content for repository [%s] on branch %s at path %s: %s", repoName, branch, resolvedPath, getErrorMessage(response)));
         }
 
-        final Optional<String> lastCommit = getLatestCommit(branch, resolvedPath);
+        return getRequiredLatestCommit(branch, resolvedPath);
+    }
 
-        if (lastCommit.isEmpty()) {
-            throw new FlowRegistryException(String.format("Could not find commit for the file %s we just tried to commit on branch %s", resolvedPath, branch));
+    private String createContentDataCenter(final GitCreateContentRequest request, final String resolvedPath, final String branch) throws FlowRegistryException {
+        final StandardMultipartFormDataStreamBuilder multipartBuilder = new StandardMultipartFormDataStreamBuilder();
+        final String fileName = getFileName(resolvedPath);
+        multipartBuilder.addPart("content", fileName, StandardHttpContentType.APPLICATION_OCTET_STREAM, request.getContent().getBytes(StandardCharsets.UTF_8));
+        multipartBuilder.addPart("branch", StandardHttpContentType.TEXT_PLAIN, branch.getBytes(StandardCharsets.UTF_8));
+
+        final String message = request.getMessage();
+        if (message != null && !message.isEmpty()) {
+            multipartBuilder.addPart("message", StandardHttpContentType.TEXT_PLAIN, message.getBytes(StandardCharsets.UTF_8));
         }
 
-        return lastCommit.get();
+        final String existingContentSha = request.getExistingContentSha();
+        if (existingContentSha != null && !existingContentSha.isEmpty()) {
+            multipartBuilder.addPart("sourceCommitId", StandardHttpContentType.TEXT_PLAIN, existingContentSha.getBytes(StandardCharsets.UTF_8));
+        }
+
+        final HttpUriBuilder uriBuilder = getRepositoryUriBuilder().addPathSegment("browse");
+        addPathSegments(uriBuilder, resolvedPath);
+        final URI uri = uriBuilder.build();
+        final byte[] requestBody = toByteArray(multipartBuilder.build());
+        final HttpResponseEntity response = this.webClient.getWebClientService()
+                .put()
+                .uri(uri)
+                .body(new ByteArrayInputStream(requestBody), OptionalLong.of(requestBody.length))
+                .header(AUTHORIZATION_HEADER, authToken.getAuthzHeaderValue())
+                .header(CONTENT_TYPE_HEADER, multipartBuilder.getHttpContentType().getContentType())
+                .retrieve();
+
+        if (response.statusCode() != HttpURLConnection.HTTP_OK && response.statusCode() != HttpURLConnection.HTTP_CREATED) {
```

Comment:
Is it possible to return either 200 or 201? It seems like it should be one or the other.

Reply:
Yes, it can be one or the other depending on potential existing content. I updated the logic to make it easier to understand.